clojure/src/info/lotharschulz/armstrong_numbers/armstrong_numbers.clj

```diff
@@ -0,0 +1,38 @@
+(ns info.lotharschulz.armstrong_numbers.armstrong_numbers
+  (:import [java.Math]))
+
+(defn get-number-of-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns the :getNumberOfDigits of n if n is an integer and greater than -1"
+  (if (zero? n)
+    1
+    (if (pos? n )
+      (+ (int (Math/log10 n )) 1)
+      nil)))
+
+(defn get-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "return number of digits of given int n"
+    (if (zero? n)
+      [0]
+      (if (pos? n)
+        (into [] (map #(Character/getNumericValue %) (str n)))
+        [])))
+
+(defn is-armstrong-number [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns true if the given number bigger than -1 is an armstrong number"
+  (if (zero? n)
+    false
+    (if (pos? n)
+      (= n (reduce + (map #(int (Math/pow % (get-number-of-digits n))) (get-digits n))))
+      false)))
+
+(defn generate-armstrong-numbers [^Integer from, ^Integer to]
+  {:pre [(not (nil? from)) (integer? from) (pos? from) (not (nil? to)) (integer? to) (pos? to) ]}
```

Comment:
* `(integer? nil) => false`, no need for `(not (nil? n))`
* `(not (nil? n))` is more nicely expressed using `(some? n)`
It makes sense to use `:pre` if would hurt not to use it. In this case providing negative `from` or `to` will not hurt, as `is-armstrong-number` will return `false` on them anyway. 
Being friendly and tolerant to the caller pays back.
clojure.core functions usually are tolerant to their inputs, returning either nil or empty seq, or false instead of throwing exceptions

```
(count nil) => 0
(range 5 1) => ()
(conj nil 1) => (1)
```
etc.